### PR TITLE
feat: work around get_stacktrace deprecation/removal

### PR DESCRIPTION
**Pull request by nickva**
This patch introduces a macro and inserts it everywhere we catch errors
and then generate a stacktrace.

So far the only thing that is a little bit ugly is that in two places,
I had to add a header include dependency on couch_db.erl where those
modules didn’t have any ties to couchdb/* before, alas. I’d be willing
to duplicate the macros in those modules, if we don’t want the include
dependency.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -187,7 +187,7 @@ exunit: export ERL_AFLAGS = -config $(shell pwd)/rel/files/eunit.config
 exunit: export COUCHDB_QUERY_SERVER_JAVASCRIPT = $(shell pwd)/bin/couchjs $(shell pwd)/share/server/main.js
 exunit: export COUCHDB_TEST_ADMIN_PARTY_OVERRIDE=1
 exunit: couch elixir-init setup-eunit elixir-check-formatted elixir-credo
-	@mix test --cover --trace $(EXUNIT_OPTS)
```
Comment (wohali) [inline Makefile:190]: unrelated?

Comment (nickva) [reply Makefile:190]: It's from Jan's cherry-picked commit

Comment (nickva) [reply Makefile]: Removed it. Good catch

**Review by wohali — APPROVED** [review]
